src/ProducerInterface.php

```diff
@@ -5,10 +5,10 @@
 interface ProducerInterface
 {
     /**
-     * @param string $topic The name of the topic you will produce the message
-     * @param string $payload The payload to be send
-     * @param array $configs Specific configurations
+     * @param  string $topic   The name of the topic / channel to send the message to
+     * @param  string $payload The payload to be sent
+     * @param  array  $configs Specific configurations
      * @return mixed
      */
-    public function produce($topic, $payload, $configs = []);
+    public function produce($topic, $payload, array $configs = []);
```

Comment:
I'd rather expose primitives to the public API - users may have their own collection class - so I'd rather remove the type hint completely rather than hint MapInterface.